src/main/java/com/salesforce/dataloader/action/visitor/BulkV2Connection.java

```diff
@@ -0,0 +1,364 @@
+        }
+
+        String encoding = connection.getHeaderField("Content-Encoding");
+        if ("gzip".equals(encoding)) {
+            in = new GZIPInputStream(in);
+        }
+
+        if (getConfig().isTraceMessage() || getConfig().hasMessageHandlers()) {
+            byte[] bytes = FileUtil.toBytes(in);
+            in = new ByteArrayInputStream(bytes);
+
+            if (getConfig().hasMessageHandlers()) {
+                Iterator<MessageHandler> it = getConfig().getMessagerHandlers();
+                while (it.hasNext()) {
+                    MessageHandler handler = it.next();
+                    if (handler instanceof MessageHandlerWithHeaders) {
+                        ((MessageHandlerWithHeaders)handler).handleRequest(url, new byte[0], null);
+                        ((MessageHandlerWithHeaders)handler).handleResponse(url, bytes, connection.getHeaderFields());
+                    } else {
+                        handler.handleRequest(url, new byte[0]);
+                        handler.handleResponse(url, bytes);
+                    }
+                }
+            }
+
+            if (getConfig().isTraceMessage()) {
+                getConfig().getTraceStream().println(url.toExternalForm());
+
+                Map<String, List<String>> headers = connection.getHeaderFields();
+                for (Map.Entry<String, List<String>>entry : headers.entrySet()) {
+                    StringBuffer sb = new StringBuffer();
+                    List<String> values = entry.getValue();
+
+                    if (values != null) {
+                        for (String v : values) {
+                            sb.append(v);
+                        }
+                    }
+
+                    getConfig().getTraceStream().println(entry.getKey() + ": " + sb.toString());
+                }
+
+                getConfig().teeInputStream(bytes);
+            }
+        }
+
+        if (!success) {
+            ContentType type = null;
+            String contentTypeHeader = connection.getContentType();
+            if (contentTypeHeader != null) {
+                if (contentTypeHeader.contains(XML_CONTENT_TYPE)) {
+                    type = ContentType.XML;
+                } else if (contentTypeHeader.contains(JSON_CONTENT_TYPE)) {
+                    type = ContentType.JSON;
+                }
+            }
+            parseAndThrowException(in, type);
+        }
+
+        this.queryLocator = connection.getHeaderField("Sforce-Locator");
+        this.queryResultNumberOfRecords = connection.getHeaderField("Sforce-NumberOfRecords");
+        return in;
+    }
+    
+    public String getQueryLocator() {
+        return this.queryLocator;
+    }
+
+    public JobInfo closeJob(String jobId) throws AsyncApiException {
+        
+        JobInfo job = new JobInfo();
+        job.setId(jobId);
+        job.setState(JobStateEnum.Closed);
+        return job;
+        /*
+        return updateJob(job);
+        */
+    }
```

Comment:
But Bulk V2 does support deleting jobs. So maybe a deleteJob() method should be added.

Reply:
Data Loader does not delete bulk jobs. So, I will not add it to BulkV2Connection.java for now.